File: src/backend/app/schemas/menu.py
# backend/app/schemas/menu.py
from typing import List, Optional

from pydantic import BaseModel, Field


class ItemBase(BaseModel):
    id: str
    category_id: str
    name_zh: str
    name_en: str
    description_zh: Optional[str] = None
    description_en: Optional[str] = None
    price_sgd: float
    image_url: Optional[str] = None
    options_config: dict = Field(default_factory=dict)
    is_available: bool = True
    stock: Optional[int] = None
    sort_order: int = 0


class ItemResponse(ItemBase):
    class Config:
        from_attributes = True


class CategoryBase(BaseModel):
    id: str
    name_zh: str
    name_en: str
    sort_order: int


class CategoryWithItems(CategoryBase):
    items: List[ItemResponse] = []


class MenuResponse(BaseModel):
    categories: List[CategoryWithItems] = []

File: src/frontend/src/types/menu.ts
// Types for Menu
export interface ItemOptions {
  size?: 'S' | 'M' | 'L'
  sweetness?: string
  temperature?: string
  extras?: string[]
}

export interface MenuItem {
  id: string
  category_id: string
  name_zh: string
  name_en: string
  description_zh?: string
  description_en?: string
  price_sgd: number
  image_url?: string
  options_config: {
    sizes?: string[]
    sweetness?: string[]
    temperature?: string[]
    extras?: { name: string; price: number }[]
  }
  is_available: boolean
  stock?: number | null
  sort_order: number
}

export interface Category {
  id: string
  name_zh: string
  name_en: string
  sort_order: number
  items: MenuItem[]
}

export interface MenuResponse {
  categories: Category[]
}
